### ISSUE TITLE
Question: 

### ISSUE THREAD

**Issue by jecampagne** [google/neural-tangents#136]
Hi, 

Let define `d` the number of features, `N` a neurone numbers and `ns` the number of training samples eg.
 `d:20, ns:200, N:200`. Here is a simple 2-layers Dense NN

```python
init_fn, apply_fn, kernel_fn = stax.serial(
    stax.Dense(N, W_std=1., parameterization='standard'), 
    stax.Relu(),
    stax.Dense(1, W_std=1., parameterization='standard')
    )

apply_fn = jit(apply_fn)
kernel_fn = jit(kernel_fn, static_argnums=(2,))
```
Some random dataset
```python
X_train = jax.random.normal(key, shape=(ns,d))
Y_train =  jax.random.normal(key, shape=(ns,1))

ns_test=1000
X_test = jax.random.normal(key, shape=(ns_test,d))
Y_test=  jax.random.normal(key, shape=(ns_test,1))

train = (X_train, Y_train)
test  = (X_test, Y_test)
```
I can now choose to train the NN with MSE loss
```python
loss_nn = jit(lambda params, x, y: 0.5 * np.mean((apply_fn(params, x) - y) ** 2))
grad_loss_nn = jit(lambda state, x, y: grad(loss_nn)(get_params(state), x, y))

learning_rate = 1e-2
training_steps = 10_000

#exp scheduler step_size * decay_rate ** (i / decay_steps)
opt_init, opt_update, get_params = optimizers.adam(
    optimizers.exponential_decay(learning_rate,decay_steps=1000,decay_rate=0.1))
opt_update = jit(opt_update)

_, params = init_fn(key, (-1, d)) 
opt_state = opt_init(params)

def train_network(key):
    train_losses = []
    test_losses = []

    _, params = init_fn(key, (-1, d)) 
    opt_state = opt_init(params)

    for i in range(training_steps):
        train_losses += [np.reshape(loss_nn(get_params(opt_state), *train), (1,))]  
        test_losses += [np.reshape(loss_nn(get_params(opt_state), *test), (1,))]
        opt_state = opt_update(i, grad_loss_nn(opt_state, *train), opt_state)

    train_losses = np.concatenate(train_losses)
    test_losses = np.concatenate(test_losses)
    return get_params(opt_state), train_losses, test_losses

params, train_loss, test_loss = train_network(key)
```

Ok so far so go. Now, if I want to study the finite-N  Neural Tangent Kernel associated to the above NN which is something like (with `\sigma(x)=ReLU(x)`: 
```
K_NT(x1,x2; N) = <x_1, x_2> 1/N \sum_j=1^N \sigma^\prime(<W_j, x_1>) \sigma^\prime(<W_j, x_2>)
```
and get the inference on X_test (ie f_NT(X_test)) which function should I call ?  (Q1)

Then, If N goes to infinity the Kernel tends to this one 
```
K_NT(x1,x2,Infinity) = <x_1, x_2> \mathbb{E}_{W}[\sigma^\prime(<W_j, x_1>) \sigma^\prime(<W_j, x_2>)]
 ```
which function should I call to get the inference on X_test? (Q2)

I am a bit puzzled in the different `nt.predict.gradient_descent_mse (nt.predict.gradient_descent_mse_ensemble)` (with get='ntk').

Thanks for your kind help.

**Comment by romanngg** [comment]
1) For finite-width, finite or infinite-time (`t=None`) MSE loss, you should use `nt.predict.predict.gradient_descent_mse`, and pass your finite-width kernels `k_train_train`/`k_test_train` computed with `nt.empirical_kernel_fn`/`nt.empirical_ntk_fn`. See example here: https://github.com/google/neural-tangents/blob/main/examples/function_space.py. You can also use `nt.predict.gradient_descent` with a custom `loss`, but this will numerically solve the ODE, so not computed in closed form / can diverge, unlike `nt.predict.gradient_descent_mse`.

2) For infinite width, finite or infinite-time, you should use `nt.predict.gradient_descent_mse_ensemble`, and additionally for infinite time, you can also use `nt.predict.gp_inference`, and pass them your `kernel_fn` defined in your code above, or the `k_train_train` computed as `kernel_fn(x_train, None, 'ntk')`. `gp_inference` is just a different (more limited, but sometimes more convenient) way of calling `gradient_descent_mse_ensemble` with `t=None`, i.e. infinite time. Note that formally this is doing inference with an infinite _ensembe_ of infinite-width networks.

Check out https://github.com/google/neural-tangents/issues/72#issuecomment-708906167 for more details.
Lmk if this helps!

**Comment by jecampagne** [comment]
Ok, nice and kind answer. Look at my last post on a concrete run to perform  some numerical simultion along the line of the authors of  https://arxiv.org/pdf/2007.12826.pdf
